addons/hr_holidays/models/hr_leave.py

```diff
@@ -916,14 +916,14 @@ def action_validate(self):
                 if holiday.leave_type_request_unit != 'day' or any(l.leave_type_request_unit == 'hour' for l in conflicting_leaves):
                     raise ValidationError(_('You can not have 2 leaves that overlaps on the same day.'))
 
+                target_states = {l.employee_id: l.state for l in conflicting_leaves}
```

Comment:
Is the employee really the right level of resolution here? Surely there can be multiple leaves for the same employee in different states?

Might be better to have a second list parallel to `split_leaves_vals` which stores the `target_state`, then we can `zip()` the old states a new vals (batch create should keep order) in order to properly partition between the desired states for the new leaves?